tools/toolchains/__init__.py

```diff
@@ -887,7 +887,10 @@ def compile_command(self, source, object, includes):
         if ext == '.c' or  ext == '.cpp':
             base, _ = splitext(object)
             dep_path = base + '.d'
-            deps = self.parse_dependencies(dep_path) if (exists(dep_path)) else []
+            try:
+                deps = self.parse_dependencies(dep_path) if (exists(dep_path)) else []
+            except:
```

Comment:
A bare "except:" is generally considered [poor practice](https://docs.python.org/2/howto/doanddont.html#except).  E.g. a well-timed Ctrl-C, instead of stopping compilation, may instead cause *more* compilation to take place.